src/main/java/com/iota/iri/service/snapshot/impl/SnapshotStateImpl.java

```diff
@@ -130,6 +135,24 @@ public SnapshotState patchedState(SnapshotStateDiff snapshotStateDiff) {
         return new SnapshotStateImpl(patchedBalances);
     }
 
+    @Override
+    public int hashCode() {
+        return Objects.hash(getClass(), balances);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+
+        if (obj == null || !getClass().equals(obj.getClass())) {
+            return false;
+        }
+
+        return Objects.equals(balances, ((SnapshotStateImpl) obj).balances);
+    }
+
```

Comment:
Technically every "model" object should have a toString() but in this case it will probably be obnoxiously long...
You can add it if you think it will help, or decide not to

Reply:
as i said in the other comment - maybe not necessarily in the scope of this PR. The equals and hashCode methods were necessary for the tests.